### PR TITLE
feat(station): use own policies & permissions for canisters cycles fund

### PR DESCRIPTION
**Pull request by keplervital**
Separates the option to send cycles to an external canister out of the `Change` operation into it's own operation and permission, this enables different Request Policies to be attached to that privileged request type.

### CODE REVIEW

File: core/station/api/spec.did [hunk 1]
```diff
@@ -679,6 +678,29 @@ type ConfigureExternalCanisterOperationInput = record {
 
 type ConfigureExternalCanisterOperation = ConfigureExternalCanisterOperationInput;
 
+// The input type for funding an external canister in the station.
```
Comment (mraszyk) [inline core/station/api/spec.did:681]:
```suggestion
// The operation kind for funding an external canister in the station.
```

File: core/station/api/spec.did [hunk 2]
```diff
@@ -871,6 +895,8 @@ type RequestOperationInput = variant {
   ConfigureExternalCanister : ConfigureExternalCanisterOperationInput;
   // An operation for calling an external canister.
   CallExternalCanister : CallExternalCanisterOperationInput;
+  // An operation for sending cycles to an external canister.
```
Comment (mraszyk) [inline core/station/api/spec.did:898]:
```suggestion
  // An operation for funding an external canister.
```
also below

File: core/station/api/spec.did [hunk 3]
```diff
@@ -1810,14 +1840,8 @@ type ChangeCanisterResourceAction = variant {
...
 
-// The target canister of a request to change a external canister.
-type ChangeExternalCanisterResourceTarget = variant {
-  Any;
-  Canister : principal;
-};
-
-// The target canister of a request to read (e.g., fetch status of) a external canister.
-type ReadExternalCanisterResourceTarget = variant {
+// The target canister of a interact with.
```
Comment (mraszyk) [inline core/station/api/spec.did:1843]:
```suggestion
// The target canister to interact with.
```

File: core/station/impl/src/mappers/resource.rs [hunk 4]
```diff
@@ -313,6 +277,9 @@ impl From<ExternalCanisterResourceAction> for station_api::ExternalCanisterResou
             ExternalCanisterResourceAction::Create => {
                 station_api::ExternalCanisterResourceActionDTO::Create
             }
+            ExternalCanisterResourceAction::Fund(target) => {
+                station_api::ExternalCanisterResourceActionDTO::Change(target.into())
```
Comment (mraszyk) [inline core/station/impl/src/mappers/resource.rs:281]:
```suggestion
                station_api::ExternalCanisterResourceActionDTO::Fund(target.into())
```

File: core/station/impl/src/models/request_specifier.rs [hunk 5]
```diff
@@ -358,14 +358,22 @@ mod tests {
...
+        RequestSpecifier::ChangeExternalCanister(ExternalCanisterId::Canister(
             Principal::management_canister(),
         ))
         .validate()
         .expect("ChangeExternalCanister should be valid");
+        RequestSpecifier::FundExternalCanister(ExternalCanisterId::Any)
+            .validate()
+            .expect("FundExternalCanister should be valid");
+        RequestSpecifier::FundExternalCanister(ExternalCanisterId::Canister(
+            Principal::management_canister(),
```
Comment (mraszyk) [inline core/station/impl/src/models/request_specifier.rs:373]: Usually, the management canister is not a valid external canister.

File: core/station/impl/src/models/resource.rs [hunk 6]
```diff
@@ -376,24 +371,46 @@ impl Resource {
...
                 }
-                ExternalCanisterResourceAction::Change(
-                    ChangeExternalCanisterResourceTarget::Any,
-                ) => {
+                ExternalCanisterResourceAction::Fund(ExternalCanisterId::Any) => {
+                    vec![
+                        Resource::ExternalCanister(ExternalCanisterResourceAction::Fund(
+                            ExternalCanisterId::Any,
+                        )),
+                        // The following additional resources also enable the user to perform the `Fund` action.
```
Comment (mraszyk) [inline core/station/impl/src/models/resource.rs:379]: Why?

Comment (keplervital) [reply core/station/impl/src/models/resource.rs:379]: Should a user that can request for changes to the canister not be able to request for it to be funded?

Comment (mraszyk) [reply core/station/impl/src/models/resource.rs:379]: Now I understand the motivation - makes sense to me!

**Review by mraszyk — APPROVED** [review]

**Pull request merged**